### ISSUE TITLE
Error caused by pydantic backwards compatibility break

### ISSUE THREAD

**Issue by arcondello** [dwavesystems/dwave-cloud-client#553]
It seems the `pydantic 2.0` introduced some backwards compatibility breaks.  We now get errors like
```
TypeError: To define root models, use `pydantic.RootModel` rather than a field called '__root__
```

See [this CI build](https://app.circleci.com/pipelines/github/dwavesystems/dwave-ocean-sdk/1810/workflows/4a71cdf9-bb87-465f-b8f5-eab0ed11b08b/jobs/26986) for a full traceback.

**Comment by arcondello** [comment]
I also get errors running with [pydantic 1.7.3](https://github.com/dwavesystems/dwave-cloud-client/blob/96db43f1e6db8cfd091050431866b97bb90c4882/setup.py#L16).

**Comment by randomir** [comment]
Quick fix in #554, released as 0.10.6. Code overhaul for v2 coming.
